iocore/eventsystem/I_MIOBufferWriter.h

```diff
@@ -0,0 +1,138 @@
+#if !defined I_MIOBUFFERWRITER_H_
+#define I_MIOBUFFERWRITER_H_
+
+/** @file
+
+    Buffer Writer for an MIOBuffer.
+
+    @section license License
+
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+ */
+
+#include <cstring>
+
+#include <ts/ink_assert.h>
+#include <ts/BufferWriter.h>
+
+#include <I_IOBuffer.h>
+
+class MIOBufferWriter : public ts::BufferWriter
+{
+public:
+  MIOBufferWriter(MIOBuffer &miob) : _miob(miob) { }
+
+  MIOBufferWriter &
+  write(const void *data_, size_t length) override
+  {
+    const char *data = reinterpret_cast<const char *>(data_);
```

Comment:
`static_cast` doesn't work?

Reply:
Probably.  I generally like to use the scariest cast possible when converting from a void pointer.  But converting to char is not scary really, so static_cast would be fine.